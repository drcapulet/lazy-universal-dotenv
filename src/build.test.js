process.env.BUILD_TARGET = "client"

const snapshotOpts = {
  APP_ROOT: expect.any(String)
}

/* eslint-disable import/no-commonjs */
// We can't use ESM when relying on the fact the the env from the top is correctly respected.
const api = require(".")

test("Serializes BUILD_TARGET", () => {
  const { raw, stringified, webpack } = api.getEnvironment()
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
})

test("Exports BUILD_TARGET", () => {
  const { raw } = api.getEnvironment()
  expect(raw.BUILD_TARGET).toBe("client")
})
