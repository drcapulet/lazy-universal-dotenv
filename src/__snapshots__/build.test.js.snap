// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serializes BUILD_TARGET 1`] = `
Object {
  "APP_ROOT": Any<String>,
  "APP_TEST_IS_CLIENT": "true",
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "BUILD_TARGET": "client",
  "NODE_ENV": "test",
}
`;

exports[`Serializes BUILD_TARGET 2`] = `
Object {
  "APP_ROOT": Any<String>,
  "APP_TEST_IS_CLIENT": "\\"true\\"",
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "BUILD_TARGET": "\\"client\\"",
  "NODE_ENV": "\\"test\\"",
}
`;
